rules/interface_locks.go

```diff
@@ -0,0 +1,172 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+var _ tflint.Rule = new(TerraformLockInterfaceRule)
+
+type TerraformLockInterfaceRule struct {
+	tflint.DefaultRule
+}
+
+func NewTerraformLockInterfaceRule() *TerraformLockInterfaceRule {
+	return new(TerraformLockInterfaceRule)
+}
+
+func (t *TerraformLockInterfaceRule) Name() string {
+	return "tfintlock"
+}
+
+func (t *TerraformLockInterfaceRule) Link() string {
+	return "https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#resource-locks"
+}
+
+func (t *TerraformLockInterfaceRule) Enabled() bool {
+	return false
+}
+
+func (t *TerraformLockInterfaceRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (t *TerraformLockInterfaceRule) Check(r tflint.Runner) error {
+	path, err := r.GetModulePath()
+	if err != nil {
+		return err
+	}
+	if !path.IsRoot() {
+		// This rule does not evaluate child modules.
+		return nil
+	}
+	body, err := r.GetModuleContent(&hclext.BodySchema{
+		Blocks: []hclext.BlockSchema{
+			{
+				Type:       "variable",
+				LabelNames: []string{"name"},
+				Body: &hclext.BodySchema{
+					Attributes: []hclext.AttributeSchema{
+						{Name: "type"},
+						{Name: "default"},
+					},
+				},
+			},
+		},
+	}, &tflint.GetModuleContentOption{ExpandMode: tflint.ExpandModeNone})
+	if err != nil {
+		return err
+	}
+
+	for _, variable := range body.Blocks {
+		if variable.Labels[0] != "lock" {
+			continue
+		}
+		// Check if the variable has a type attribute and that it is correct.
+		typeattr, exists := variable.Body.Attributes["type"]
+		if !exists {
+			if err := r.EmitIssue(
+				t,
+				fmt.Sprintf("`%s` variable type not declared", variable.Labels[0]),
+				variable.DefRange,
+			); err != nil {
+				return err
+			}
+			continue
+		}
+		if err := checkLockInterface(typeattr); err != nil {
+			if err2 := r.EmitIssue(
+				t,
+				fmt.Sprintf("`var.%s`: %s", variable.Labels[0], err.Error()),
+				variable.DefRange,
+			); err2 != nil {
+				return err2
+			}
+		}
+
+		defaultattr, exists := variable.Body.Attributes["default"]
+		if !exists {
+			if err := r.EmitIssue(
+				t,
+				fmt.Sprintf("`var.%s`: default not declared", variable.Labels[0]),
+				variable.DefRange,
+			); err != nil {
+				return err
+			}
+			continue
+		}
+		defaultval, _ := defaultattr.Expr.Value(nil)
+		if !defaultval.IsNull() {
+			if err := r.EmitIssue(
+				t,
+				fmt.Sprintf("`var.%s`: default value is not `null`", variable.Labels[0]),
+				variable.DefRange,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func checkLockInterface(attr *hclext.Attribute) error {
+	expr, ok := attr.Expr.(*hclsyntax.FunctionCallExpr)
```

Comment:
About how to check whether the `type` is what we want, I'm thinking of, what if we evaluate this attribute's value, what would this value looks like? I assume it would be an object with two attributes, `kind` and `name`.

Reply:
This is how to look up the "string" value from the "type = string" object, however the .data property is private and we cannot access it from this package.

`(*(*(*(*attr).Expr.(data)).Args[0].(data)).Items[0].ValueExpr.(data)).Traversal[0].(data).Name`

Reply:
This seems like the package we need:

https://pkg.go.dev/github.com/hashicorp/hcl/v2/ext/typeexpr#section-readme